solr/core/src/test/org/apache/solr/cloud/DeleteShardTest.java

```diff
@@ -125,4 +131,19 @@ private void setSliceState(String collectionName, String shardId, Slice.State st
         collectionName,
         c -> c.getSlice(shardId).getState() == state);
   }
+
+  /** Check whether shard metadata exist in Zookeeper. */
+  private void assertShardMetadata(String collection, String sliceId, boolean shouldExist)
```

Comment:
could this be improved to scan the ZK nodes for any ZNode that matches the shard ID?  This way, new/unexpected stuff could be detected.